src/components/ShoppingList/ShoppingList.js

```diff
@@ -1,28 +1,78 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { deleteList } from '../../actions/shoppingLists';
 import '../component.css';
 
-export default function ShoppingList({ id, name, store, editing }) {
-  if (editing) {
+export class ShoppingList extends Component {
+  // export default function ShoppingList({ id, name, store, editing }) {
+  // if (editing) {
+  //   return (
+  //     <li key={id}>
+  //       <form id={id} />
+  //       <input name="name" defaultValue={name} />
+  //       {store ? <input name="store" defaultValue={store.name} /> : null}
+  //       <button>delete</button>
+  //     </li>
+  //   );
+  // }
+  constructor(props) {
+    super(props);
+    this.state = {
+      deleteModal: false,
+    };
+  }
+
+  deleteClicked() {
+    this.setState({
+      deleteModal: !this.state.deleteModal,
+    });
+    console.log(this.state);
+  }
+
+  confirmDelete() {
+    this.props.deleteList(this.props.id);
+    this.setState({
+      deleteModal: !this.state.deleteModal,
+    });
+  }
+
+  render() {
+    const { id, store, name } = this.props;
```

Comment:
So, I don't know why this doesn't error out in the edit-list-name branch, but I get a React-Redux error that passing the redux store is not allowed.

This is due to the fact that React-Redux treats the `store` prop as a reserved word. As result, you will have to use a different name (`groceryStore`?) for the prop. If you don't want to change your code too much, change this line to

```javascript
const { id, groceryStore: store, name } = this.props;
```

Just be sure to change the prop that is passed to this component in Lists.js.

Reply:
thanks!